scene/resources/animation.cpp

```diff
@@ -4154,11 +4154,12 @@ bool Animation::_float_track_optimize_key(const TKey<float> t0, const TKey<float
 }
 
 bool Animation::_vector2_track_optimize_key(const TKey<Vector2> t0, const TKey<Vector2> t1, const TKey<Vector2> t2, real_t p_allowed_velocity_err, real_t p_allowed_angular_error, real_t p_allowed_precision_error, bool p_is_nearest) {
+	const real_t allowed_precision_error_squared = p_allowed_precision_error * p_allowed_precision_error;
 	// Remove overlapping keys.
 	if (Math::is_equal_approx(t0.time, t1.time) || Math::is_equal_approx(t1.time, t2.time)) {
 		return true;
 	}
-	if ((t0.value - t1.value).length() < p_allowed_precision_error && (t1.value - t2.value).length() < p_allowed_precision_error) {
+	if ((t0.value - t1.value).length_squared() < allowed_precision_error_squared && (t1.value - t2.value).length_squared() < allowed_precision_error_squared) {
```

Comment:
These can all probably be replaced with:
```suggestion
	if (t0.value.distance_squared_to(t1.value) < allowed_precision_error_squared && t1.valuedistance_squared_to(t2.value) < allowed_precision_error_squared) {
```
But left them as is for now to reduce the changes